### PR TITLE
[TOPI][x86] Cascade lake support.

### PR DESCRIPTION
**Pull request by anijain2305**
As Title. Enables Intel VNNI instructions for Int8 quantized networks.

### CODE REVIEW

**Comment by anijain2305** [comment]
> Nit: why do we have the intrinsics named {`dot_16x1x16_uint8_int8_int32_skylake`, `dot_16x1x16_uint8_int8_int32_vnni`} instead of {`dot_16x1x16_uint8_int8_int32_skylake`, `dot_16x1x16_uint8_int8_int32_cascadelake`}?

Will make it more consistent. Thanks for the suggestion :)

File: tests/python/relay/test_op_level2.py [hunk 1]
```diff
@@ -576,57 +576,76 @@ def _compile(ic, oc, target, data_layout, kernel_layout, dtypes):
...
+        intel_device_type = None
+        if 'skylake-avx512' in target:
+            return "pmaddubs" in asm
+        elif 'cascadelake' in target:
+            return "vpdpbusd" in asm
+        else:
+            assert False, "Target should be Skylake or Cascadelake"
+
+    # compile conv2d for x86 (skylake, cascadelake) and test assembly contains *pmadd* instructions
+    # targets = ["llvm -mcpu=skylake-avx512", "llvm -mcpu=cascadelake"]
```
Comment (tmoreau89) [inline tests/python/relay/test_op_level2.py]: why is this commented out? do we not want to test the `cascadelake` target?

Comment (anijain2305) [reply tests/python/relay/test_op_level2.py]: Sorry, Let me add WIP tag. Jenkins is failing w/o any debug prints. So I am trying to narrow down the issue. I will ping you once I identify the issue.

File: topi/python/topi/x86/tensor_intrin.py [hunk 2]
```diff
@@ -19,15 +19,34 @@
 import tvm
 
 
-def dot_16x1x16_int8_int8_int32():
+def dot_16x1x16_uint8_int8_int32():
+    """Dispatch the most optimized intrin depending on the target"""
+    target = tvm.target.current_target()
+    intel_device_type = None
+    for opt in target.options:
+        if opt == '-mcpu=skylake-avx512':
```
[the remark this answers is not in the document]
Comment (anijain2305) [reply topi/python/topi/x86/tensor_intrin.py]: I agree. Let me see if I can get that parsing a part of target data structure.

**Comment by anijain2305** [comment]
@tmoreau89 @soiferj I am done now. Guarded the tests to work with only >=8 LLVM version. That resolved the CI error.

Please review again and let me know if you have more comments.

File: python/tvm/target.py [hunk 3]
```diff
@@ -521,6 +521,28 @@ def create(target_str):
     return _api_internal._TargetFromString(target_str)
 
 
+def get_mcpu(target):
```
Comment (soiferj) [inline python/tvm/target.py]: Is it possible to add this as a property under the Target class? Like `libs`

Comment (anijain2305) [reply python/tvm/target.py]: You mean here 

https://github.com/dmlc/tvm/blob/02c1e11716a5afdfc0159a1b21e45a64f7875473/python/tvm/target.py#L84-L136


I was planning to, but then I saw that tvm.target.create is a separate function. I did not know why that was not a part of Target data structure. So, I decided to add as a helper. 

Do let me know if you have a better suggestion.

Comment (soiferj) [reply python/tvm/target.py]: LGTM

**Pull request merged**